### ISSUE TITLE
FTPClient should support removeDirectory

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5259]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#5259|
|Type|enhancement|
|Created|2011-09-09 16:14:30Z|
|Branch|https://github.com/twisted/twisted/tree/ftp-rmd-5259|

The RMD FTP command was added in RFC959 and is implemented by many servers http://tools.ietf.org/html/rfc959

The RFC says that RMD has the same behavior as for DELE.

I have added a removeDirectory method to FTPClient in a similar way as removeFile.

I will soon attach the branch.

<details><summary>Searchable metadata</summary>

```
trac-id__5259 5259
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__ 
branch__branches_ftp_rmd_5259 branches/ftp-rmd-5259
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__ftp ftp
keywords__ 
time__1315584870000000 1315584870000000
changetime__1316377837000000 1316377837000000
version__None None
owner__ 
cc__itamarst
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban set status to `assigned`|
|-|-|

I have pushed a branch and created a Merge Proposal on LP to help see the diff.


Branch:
https://code.launchpad.net/~adiroiban/twisted/twisted-5259

MP:
https://code.launchpad.net/~adiroiban/twisted/twisted-5259/+merge/74827

Please let me know what should be the next step for this ticket.

Cheers,
Adi

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner<br>@exarkun set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst commented|
|-|-|

(In [32539]) Branching to 'ftp-rmd-5259'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst set status to `closed`|
|-|-|

(In [32544]) Merge ftp-rmd-5259.

Author: Adi Roiban
Fixes: [#5259](https://github.com/twisted/twisted/issues/5259)

Add support for removing directories to the FTP client.
